pkg/streamingpromql/query.go

```diff
@@ -37,6 +37,7 @@ import (
 var errQueryCancelled = cancellation.NewErrorf("query execution cancelled")
 var errQueryClosed = cancellation.NewErrorf("Query.Close() called")
 var errQueryFinished = cancellation.NewErrorf("query execution finished")
+var errStringForRangeQuery = errors.New("query expression produces a string, but expression for range queries must produce an instant vector or scalar")
```

Comment:
(nit) could make this generic to use in newQuery too
```suggestion
var strStringForRangeQuery = "query expression produces a %s but expression for range queries must produce an instant vector or scalar"
```

Reply:
I've tackled this in a slightly different way in f178f97: I've changed `Exec` to not check if it's an instant query if we get a `StringOperator`. This makes the behaviour consistent with how we handle `RangeVectorOperator`s, which are also not supported as the root operator for instant queries.